src/api/erc20/Erc20ApiMock.ts

```diff
@@ -82,16 +80,47 @@ export class Erc20ApiMock implements Erc20Api {
     this._initBalances(fromAddress, tokenAddress)
     this._initBalances(toAddress, tokenAddress)
 
+    // Since we don't have a msg.sender, we mimic it by using the optional field
+    const { senderAddress: spenderAddress } = txOptionalParams
```

Comment:
I'm not sure it's a good idea to add this parameter which will have to go when switching to real contracts, anyway.
As `msg.sender` is the internal responsibility of Mmask/other provider, how about grabbing it from a const
```
class Erc20ApiMock ...
  static sender = ....
  ...
  const sender = this.sender || userAddress
```
or something

Also in real web3 TxParams it's called `from`, not that it matters yet.

Reply:
I thought about this, and in the one hand I agree with @DmitryBespalov , web3 usually have a default account setup and you can define any other by changing the `from`. But in the other, i thought it's more general if we provide the address (this is precisely what I was going to review in #105 ). If you see the other APIs, I send always the `fromAddress`.

![image](https://user-images.githubusercontent.com/2352112/66116409-f8b4c480-e5d2-11e9-8601-97431503041d.png)

The reason I did this is to make APIs more general. This also allows in the future for example to reuse this APIs easily for any backend service, bots or CLI that handle multiple accounts.

Regarding the name, I like `senderAddress` and could leave it like that perfectly, but I like slightly more the `fromAddress`.

Reply:
Mmm @alfetopito, on a deeper look, I saw there was already a `fromAddress` that it's exactly that:
![image](https://user-images.githubusercontent.com/2352112/66116994-00289d80-e5d4-11e9-8131-68db4fc32878.png)

I do expect to add more optional params in `TxOptionalParams`, like maybe `gasLimit` or `gasPrice`. The `fromUser` could go, but for now I think it has enough relevance to have it's own mandatory param (I might change my mind after my #105 proposal :P )

Reply:
First, replying to the last comment by @anxolin 

> Mmm @alfetopito, on a deeper look, I saw there was already a fromAddress that it's exactly that:

This `fromAddress`, AFAIU, is inline with the `transferFrom` contract interface defined for ERC20: `function transferFrom(address _from, address _to, uint256 _value) public returns (bool success)
`

That means, I'm transferring from account `fromAddress` to account `toAddress`. The sender is not part of the parameters.

Reply:
@Velenir if I understand correctly, you mean to have the `msg.sender` as a member variable o Erc20ApiMock class?
If so, yes, that would simplify things. The sender will always be the same (the mock contract address).

I'll changed it.

Reply:
@alfetopito , no I mean the ERC20 function called `transfer`, not `transferFrom`.
`transferFrom` although is in the ERC20 standard, I think we don't need it, so I didn't create it.

So my field `fromAddress` is what dima is calling `from`.
I understand that he is meaning that that is implicit, cause most of the times in the `web3` has already a defined default account. But I model the interfaces more explicit, so you provide always the user. This is part of the review I want to do in #105

Reply:
@anxolin  Ah, so we are not using `transferFrom` at all. In this case, why do we need the allowance? 
I understood this is only used when another account is transferring from your balance on your behalf. If you yourself is calling `transfer`, there's no need to check the allowance at all, no?

Reply:
>  I think we don't need it, so I didn't create it.
Okay, I thought that, but you are actually right. It's true that we don't need it in the app itself, **BUT** you were using it for implementing the deposit mock function. This function DO need to use `transferFrom` so I guess we do have to implement it after all.

Sorry for missleading you here. I guess we need a new method in `Erc20Api` similar to:
```
transferFrom(
    tokenAddress: string,
    senderAddress: string,
    fromAddress: string,
    toAddress: string,
    amount: BN,
    txOptionalParams?: TxOptionalParams,
  ): Promise<TxResult<boolean>>
```

Reply:
Checked the base contract. It does use `transferFrom` on deposit https://github.com/gnosis/dex-contracts/blob/8c93796971b02853b2dc7dfb8332f7b97d846777/contracts/EpochTokenLocker.sol#L49-L54

I'll add it